packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/sqs.js

```diff
@@ -263,14 +265,14 @@ class InstanaAWSSQS extends InstanaAWSProduct {
               setImmediate(() => this.finishSpan(null, span));
             } else if (data && data.Messages && data.Messages.length > 0) {
               const messages = data.Messages;
-
               let tracingAttributes = readTracingAttributes(messages[0].MessageAttributes);
               if (!hasTracingAttributes(tracingAttributes)) {
                 tracingAttributes = readTracingAttributesFromSns(messages[0].Body);
               }
               if (tracingAttributes.level === '0') {
                 cls.setTracingLevel('0');
                 setImmediate(() => span.cancel());
+                return data;
```

Comment:
TBD: IMO there was a return missing here. Why would we call the logic below for a cancelled span.

Reply:
👍 👍 

Yes, I think returning here is better. Looking at this, I think if we add both `span.cancel` and `this.finishSpan` to the next tick of the event loop via setImmediate, I am not even sure if the behavior would be deterministic. If we happen to call span.cancel first, everything is fine and the `this.finishSpan` should be ignored I think. If it happens the other way around, we would create a span even though it should have been supressed.

I think the callback case in line 236 probably has the same potential issue. There, we cannot simply return, though, we would have to call `_callback.apply(self, arguments);` and then we can return.

WDYT?

Reply:
https://github.com/instana/nodejs/pull/860/commits/5c5543af89fbccc44b615a4f7a06ccdfcc2cac04

Reply:
@aryamohanan Did you review https://github.com/instana/nodejs/commit/5c5543af89fbccc44b615a4f7a06ccdfcc2cac04? That would be helpful :)